templates/php/src/Services/Service.php.twig

```diff
@@ -55,6 +55,49 @@ class {{ service.name | caseUcfirst }} extends Service
         }
 
 {% endfor %}
+{% if 'multipart/form-data' in method.consumes %}
+{% for parameter in method.parameters.all %}
+{% if parameter.type == 'file' %}
+        $size = filesize(${{ parameter.name | caseCamel }});
+        $mimeType = mime_content_type(${{ parameter.name | caseCamel }});
+        $postedName = basename(${{ parameter.name | caseCamel }});
+        //send single file if size is less than or equal to 5MB
+        if ($size <= Client::CHUNK_SIZE) {
+            $params['{{ parameter.name }}'] = new \CURLFile(${{ parameter.name | caseCamel }}, $mimeType, $postedName);
+            return $this->client->call(Client::METHOD_{{ method.method | caseUpper }}, $path, [
+{% for param in method.parameters.header %}
+                '{{ param.name }}' => ${{ param.name | caseCamel }},
+{% endfor %}
+{% for key, header in method.headers %}
+                '{{ key }}' => '{{ header }}',
+{% endfor %}
+                ], $params);
+        } else {
+            $id = '';
+            $handle = @fopen(${{ parameter.name | caseCamel }}, "rb");
+            $counter = 0;
+            $headers = ['content-type' => 'multipart/form-data'];
+            while (!feof($handle)) {
+                $params['{{ parameter.name }}'] = new \CURLFile('data://' . $mimeType . ';base64,' . base64_encode(@fread($handle, Client::CHUNK_SIZE)), $mimeType, $postedName);
+                $headers['content-range'] = 'bytes ' . ($counter * Client::CHUNK_SIZE) . '-' . min(((($counter * Client::CHUNK_SIZE) + Client::CHUNK_SIZE) - 1), $size) . '/' . $size;
+                if(!empty($id)) {
+                    $headers['x-appwrite-id'] = $id;
+                }
+                $response = $this->client->call(Client::METHOD_POST, $path, $headers, $params);
+                $counter++;
+                if(empty($id)) {
+                    $id = $response['$id'];
+                }
+                if($onProgress !== null) {
+                    $onProgress(min($offset, $size) / $size * 100);
```

Comment:
Are we sure we want to only return percentage? Let's say someone wanted to display "12 chunks left", he cannot do that.. Because he never have access to Client::CHUNK_SIZE, so even if he wanted, he can't. Should we return data object instead? For example:

```
{
  "progress": 58.777192290628086,
  "chunksTotal": 4,
  "chunksUploaded": 2,
  "sizeTotal": 17839845,
  "sizeUploaded": 10485760
}
```

Reply:
If we decide to use a more informative object, then we might want to execute onProgress also if the file is <5MB, so the frontend has this information.

Reply:
This can be a good Idea. However, for less than 5MB not sure how we can handle progress for each framework, haven't been able to figure anything similar for Dart/Flutter